File: model/transaction.go
package model

type TransactionRequest struct {
	ID          string `json:"id"`
	Amount      int    `json:"amount"`
	Date        string `json:"date"`
	Description string `json:"description"`
	UserId      string `json:"userId"`
}

type TransactionResponse struct {
	ID          string       `json:"id"`
	Amount      int          `json:"amount"`
	Date        string       `json:"date"`
	CreatedOn   string       `json:"createdOn"`
	Description string       `json:"description"`
	User        UserResponse `json:"user"`
}

type TransactionHookRequest struct {
	Amount      int          `json:"amount"`
	Description string       `json:"description"`
	Fullname	string       `json:"fullname"`
	Email       string       `json:"email"`
	Phone       string       `json:"phone"`
}

File: cms/src/models/transaction.ts
export type Transaction = { 
    id: string,
    amount: number,
    date: string,
    createdOn: string,
    description: string,
    user: User,
}

export type User = {
    id: string,
    email: string,
    fullname: string,
    phone: string,
}
